File: algoritmos de treino/outros/quickSort.py
def quicksort(array):
    # caso base, array com 1 ou 0 elementos não precisam ser "sorteados"
  if len(array) < 2:
    return array
  else:
    # caso recurssivo
    pivo = array[0]
    # sub-arry de todos os elementos que são menor que o pivo
    menor = [i for i in array[1:] if i <= pivo]
    # sub-array de todos os elementos que são maior que o pivo
    maior = [i for i in array[1:] if i > pivo]
    return quicksort(menor) + [pivo] + quicksort(maior)

print(quicksort([10, 5, 2, 3]))

File: algoritmos de treino/outros/quickSort.rb
def quicksort(array)
    if array.length < 2
      # caso base, array com 1 ou 0 elementos não precisam ser "sorteados"
      array
    else
      # caso recurssivo
      pivo = array[0]
      # sub-array de todos os elementos menos o array
      rest = array[1..-1]
      menor = rest.select { |i| i <= pivo } # subp-arry de todos os elementos que são menor que o pivo
      greater = rest.select { |i| i > pivo } # sub-array de todos os elementos que são maior que o pivo
      quicksort(menor) + [pivo] + quicksort(greater)
    end
  end
  
  print quicksort([10, 5, 2, 3])